src/main/java/org/jabref/gui/linkedfile/DownloadLinkedFileAction.java

```diff
@@ -100,7 +100,6 @@ public void execute() {
 
         Optional<Path> targetDirectory = databaseContext.getFirstExistingFileDir(filePreferences);
         if (targetDirectory.isEmpty()) {
-            dialogService.showErrorDialogAndWait(Localization.lang("Download file"), Localization.lang("File directory needs to be set or does not exist.\nPlease save your library and check the preferences."));
```

Comment:
Add a LOGGER.warn here. Non- localized.

LOGGER.warn("File directory not available while downloading {}. Storing as URL in file field.", url)

Reply:
Done.